### PR TITLE
Encrypt wallet

### PR DESCRIPTION
**Pull request by Krigpl**
Use password encrypted storage for ethereum wallet.
Using constant password right now just to split the changes into smaller PRs. Next PR will grab the password from RPC.

### CODE REVIEW

File: golem/node.py [hunk 1]
```diff
@@ -88,6 +88,8 @@ def _start_keys_auth(self) -> Deferred:
             KeysAuth,
             datadir=self._datadir,
             private_key_name=PRIVATE_KEY,
+            # TODO: user provided password
+            password='verystrongpasswodmuchsecurity',
```
Comment (etam) [inline golem/node.py:92]: doge

**Comment by Krigpl** [comment]
fwiw, this PR alone is ready for review and even merging.

File: tests/golem/core/test_keysauth.py [hunk 2]
```diff
@@ -100,12 +126,14 @@ def test_get_difficulty(self):
         assert KeysAuth.is_pubkey_difficult(ek.public_key, difficulty)
         assert KeysAuth.is_pubkey_difficult(ek.key_id, difficulty)
 
+    @freeze_time("2017-11-23 11:40:27.767804")
     @patch('golem.core.keysauth.logger')
-    def test_key_recreate_on_increased_difficulty(self, logger):
+    def test_key_recreate_on_increased_difficulty__and_backup(self, logger):
```
Comment (etam) [inline tests/golem/core/test_keysauth.py]: minor thing: `test_key_recreate_and_backup_on_increased_difficulty`

Comment (etam) [reply tests/golem/core/test_keysauth.py]: or even better: `test_key_backup_and_recreate_on_increased_difficulty`

**Review by etam — APPROVED** [review]
sehr gut

File: golem/core/variables.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 #       VARIABLES          #
 ############################
 APP_NAME = "Brass Golem"
-PRIVATE_KEY = "golem_private_key.peb"
+PRIVATE_KEY = "golem_wallet_const_password.json"
```
Comment (tworec) [inline golem/core/variables.py:11]: this enforce all beta-users to regenerate their keys... Can't we just read old key, and repack it into new file for first time?

File: golem/core/keysauth.py [hunk 4]
```diff
@@ -68,7 +95,7 @@ class KeysAuth:
     key_id = ""  # type: str
     ecc = None  # type: ECCx
 
-    def __init__(self, datadir: str, private_key_name: str,
+    def __init__(self, datadir: str, private_key_name: str, password: str,
```
Comment (tworec) [inline golem/core/keysauth.py:98]: password doc is missing

Comment (etam) [reply golem/core/keysauth.py:98]: or rename it to `key_password` - no doc needed

File: golem/core/variables.py (hunk 3 above)
Comment (etam) [reply golem/core/variables.py:11]: We're introducing the key difficulty thing, so most keys will be regenerated anyway.

Comment (Krigpl) [reply golem/core/variables.py:11]: Too much hassle for too little benefit.

Comment (etam) [reply golem/core/variables.py:11]: I think, that as long as we're on testnet, we don't need to care so much. On mainnet such feature would be mandatory.

Comment (tworec) [reply golem/core/variables.py:11]: ok

**Pull request merged**